### PR TITLE
Adding support for client, lister, informer, deepcopy generation

### LINKED ISSUE #37: Adding support for client, lister, informer, deepcopy generation
**Issue/Feature Description:**

Adding support for client, lister, informer, deepcopy generation for supported CRDs

**Why this issue to fixed / feature is needed(give scenarios or use cases):**

The issue can handle autogeneration of client code

**How to reproduce, in case of a bug:**
NA

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by AmitRoushan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

 /kind enhancement

**What this PR does / why we need it**:

The PR adds support for client, lister, informer and deepcopy generation for CRDs

**Which issue(s) this PR fixes**:

Fixes #37 

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

/kind NOT-TESTED

**Test Report**:

NA

**Special notes for your reviewer**:

The generated codes are not added in PR.

### CODE REVIEW

**Comment by AmitRoushan** [comment]
@sushanthakumar @PravinRanjan10 @joseph-v 
PTAL

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# TODO: This symlink should be relative.
+if [[ ! -e "${go_pkg_dir}" || "$(readlink "${go_pkg_dir}")" != "${ROOT_DIR}" ]]; then
+  ln -snf "${ROOT_DIR}" "${go_pkg_dir}"
+fi
+
+# generate the code with:
+cd ${ROOT_DIR}
+bash "${CODE_GEN_PATH}/code-generator/generate-groups.sh" \
+	all \
+  github.com/soda-cdm/kahu/controllers/generated/client \
```
Comment (sushanthakumar) [inline hack/update-codegen.sh:52]: we can keep generated code at "github.com/soda-cdm/kahu/client" for generic use

**Review by sushanthakumar — APPROVED** [review]

File: hack/update-codegen.sh (hunk 1 above)
Comment (AmitRoushan) [reply hack/update-codegen.sh]: Updated as per comment

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**